### PR TITLE
Fix building for ios

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/futils.cpp [hunk 1]
```diff
@@ -38,6 +38,10 @@ namespace fs = std::experimental::filesystem;
 #include <unistd.h>  // for getpid()
 #endif
 
+#if __has_include(<mach-o/dyld.h>)
+#include <mach-o/dyld.h>
```
Comment (m-kuhn) [inline src/futils.cpp:42]:
```suggestion
#include <mach-o/dyld.h>  // for _NSGetExecutablePath()
```

**Comment by neheb** [comment]
this test failure seems related to **[v1.1.0](https://github.com/google/brotli/releases/tag/v1.1.0)**

File: src/futils.cpp [hunk 2]
```diff
@@ -369,6 +373,14 @@ std::string getProcessPath() {
     auto path = fs::read_symlink(Internal::stringFormat("/proc/%d/path/a.out", getpid()));
 #elif defined(__unix__)
     auto path = fs::read_symlink("/proc/self/exe");
+#elif defined(__APPLE__)
```
Comment (neheb) [inline src/futils.cpp:376]: does this work on macOS as well? If so, might as well replace the previous condition.

Comment (m-kuhn) [reply src/futils.cpp:376]: It probably does ?!
Replace or switch positions? Not sure which other unixes use this branch.

Comment (neheb) [reply src/futils.cpp:376]: I believe PROC_PIDPATHINFO_MAXSIZE is only for macOS.

Comment (m-kuhn) [reply src/futils.cpp:376]: I'll move it up then 👍

**Review by neheb — APPROVED** [review]

**Comment by kmilos** [comment]
Backport?

**Comment by m-kuhn** [comment]
> Backport?

PR opened for your convenience

**Pull request merged**